eng/common/scripts/Verify-Resource-Ref.ps1

```diff
@@ -0,0 +1,40 @@
+
+. (Join-Path $PSScriptRoot common.ps1)
+Install-Module -Name powershell-yaml -RequiredVersion 0.4.1 -Force -Scope CurrentUser
+$ymlfiles = Get-ChildItem $RepoRoot | Where-Object {$_ -like '*.yml'}
+
+foreach ($file in $ymlfiles)
+{
+  Write-Host "Verifying '${file}'"
+  $ymlContent = Get-Content $file.FullName -Raw
+  $ymlObject = ConvertFrom-Yaml $ymlContent -Ordered
+
+  if ($ymlObject.Contains("resources"))
+  {
+    
+    if ($ymlObject["resources"]["repositories"])
+    {
+      $affectedRepos = @()
+      $repositories = $ymlObject["resources"]["repositories"]
+      foreach ($repo in $repositories)
+      {
+        $repoName = $repo["repository"]
+        if (-not ($repo.Contains("ref")))
+        {
+          $affectedRepos.Add($repoName)
+        }
+      }
+
+      if ($affectedRepos.Count -gt 0)
+      {
+          Write-Error "Ref not found in the following Repository Resources."
+          foreach ($repo in $affectedRepos)
+          {
+              Write-Output $repo
+          }
+          Write-Information "Please ensure you add a Ref: when using repository resources"
+          Write-Information "More Info https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/281/Engineering-Systems-Tools?anchor=versioning"
```

Comment:
looks like we should consider creating an ms.aka link for this. Perhaps `https://ms.aka/azsdk/engsys/tools-versioning` or something similar